onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc

```diff
@@ -399,12 +407,15 @@ uint32_t ModelBuilder::AddOperandFromPersistMemoryBuffer(
 }
 
 void ModelBuilder::AddOperations() {
-  for (const auto& node : model_proto_.graph().node()) {
-    if (auto* opBuilder = GetOpBuilder(node)) {
-      opBuilder->AddToModelBuilder(*this, node);
+  const auto& node_indices = graph_view_.GetNodesInTopologicalOrder();
+  for (size_t i = 0; i < node_indices.size(); i++) {
+    const auto* node(graph_view_.GetNode(node_indices[i]));
+    ORT_ENFORCE(nullptr != node, "node should not be null");
+    if (auto* opBuilder = GetOpBuilder(*node)) {
+      opBuilder->AddToModelBuilder(*this, *node);
     } else {
       throw std::invalid_argument(
-          "Node not supported" + node.name());
+          "Node not supported" + node->Name());
     }
```

Comment:
Name() won't always have a value as it's not required in the onnx spec and some converters don't add names. Good to include the OpType in the message. 

Please use ORT_THROW as well for consistency.